doc/design_AsyncNDRegisterAcessor_and_NumericAddressedBackend.dox

```diff
@@ -0,0 +1,158 @@
+namespace ChimeraTK {
+/**
+\page design_AsyncNDRegisterAccessor_NumericAddressed Design: AsyncNDRegisterAccessor
+
+\section  design_AsyncNDRegisterAccessor The AsyncNDRegisterAccessor and the AsyncAccessorManager
+
+The AsyncNDRegisterAccessor is the NDRegisterAccessor implementation for push-type data. It is used
+if accessors with AccessMode::wait_for_new_data are requested. See \ref transferElement_B_8
+"Technical specification: TransferElement B.8" for the requirements it is based on.
+
+The AsyncNDRegisterAccessor is a generic implementation for all backends and contains the following components
+
+- an NDRegisterAccessor<UserType>::Buffer which serves as a send buffer and contains the data value
+  (2D vector of UserType), a VersionNumber and a DataValidity flag.
+- a lock free queue to transport the data of the send buffer
+- functions to activate/deactivate the accessor, and to push data and exceptions into the queue
+- a (synchronous) accessor which is used for write operations (optional)
+
+One AsyncAccessorManager is used per interrupt that can occur. The AsyncAccessorManager has three main functions:
+
+- It serves as a factory for AsyncNDRegisterAccessors
+- It contains a list of created asynchronous accessors and a subscribe/unsubscribe mechanism
+- It provides functions to act on all asynchronous accessors
+
+\subsection AsyncNDRegisterAccessor_details Design decisions and implementation details
+
+- The data transport queue is of type UserType, so the potentially expensive and slow operations (data readout,
```

Comment:
`The data transport queue is of type UserType` - this is certainly wrong. I guess it is of the type `NDRegisterAccessor<UserType>::Buffer` (if I understand correctly, see above).

Reply:
You probably mean, the data transport queue contains data already converted to the UserType, right?